File: 2nd-Year/Data_Structures_and _Algorithms/Prefinals/StudentList/student_list.py
def main() -> None:
    CLASSMATE_COUNT = 3
    students = {}
    order_of_keys = []
    for i in range(1, CLASSMATE_COUNT + 1):
        key = input(f"Enter student number {i}: ")
        students[key] = input(f"Enter first name {i}: ")
        order_of_keys.append(key)
    print_student_list(students, order_of_keys)

    del students[order_of_keys[2]]
    user_student_number = input("Enter your student number: ")
    students[user_student_number] = input("Enter your first name: ")
    order_of_keys[2] = user_student_number
    print_student_list(students, order_of_keys)


def print_student_list(students: dict, order_of_keys: list) -> None:
    print("Student List:")
    for key in order_of_keys:
        print(f"{key} {students[key]}")


if __name__ == "__main__":
    main()

File: 2ndYear/Data_Structures_and _Algorithms/Prefinals/StudentList/StudentList.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class StudentList {
    private static final Scanner sc = new Scanner(System.in);
    private static Map<String, String> students = new HashMap<>();
    private static final int CLASSMATE_COUNT = 3;
    private static String[] orderOfKeys = new String[CLASSMATE_COUNT];

    public static void main(String[] args) {
        for (int i = 1; i <= CLASSMATE_COUNT; i++) {
            String key = input(String.format("Enter student number %s: ", i));
            students.put(key, input(String.format("Enter first name %s: ", i)));
            orderOfKeys[i - 1] = key;
        }

        printStudentList();
        students.remove(orderOfKeys[2]);
        String userStudentNum = input("Enter your student number: ");
        students.put(userStudentNum, input("Enter your first name: "));
        orderOfKeys[2] = userStudentNum;
        printStudentList();
    }

    private static void printStudentList() {
        System.out.println("Student List:");
        for (String key : orderOfKeys) {
            System.out.printf("%s %s\n", key, students.get(key));
        }
    }

    private static String input(String prompt) {
        System.out.print(prompt);
        return sc.nextLine();
    }
}